src/robust.jl

```diff
@@ -0,0 +1,108 @@
+# Robust Statistics
+
+#############################
+#
+#   Trimming outliers
+#
+#############################
+
+# Trimmed set
+"""
+    trim(x, p=0.2)
+
+Remove a proportion `p` of the highest elements, and `p` of the lowest
+elements of vector `x` and return the result. To compute the trimmed
+mean of `x`, use `mean(trim(x, p))`; to compute the variance of this
+quantity, use `trimvar(x, p)`.
+
+# Example
+```julia
+julia> trim([1,2,3,4,5], 0.2)
+ 2
+ 3
+ 4
+```
+"""
+trim(x::AbstractVector, p::Real=0.2) = trim!(copy(x), p)
+
+"""
+    trim!(x, p=0.2)
+
+A variant of `trim` that modifies `x` by removing the proportion `p`
+of the highest and lowest elements of `x`.
+"""
+function trim!(x::AbstractVector, p::Real=0.2)
+    n = length(x)
+    n > 0 || throw(ArgumentError("x can not be empty."))
+    0 <= p < 0.5 || throw(ArgumentError("p must be in 0 <= p < 0.5."))
+    g = floor(Int, n * p)
+    
+    select!(x, (n-g+1):n)
+    select!(x, 1:g)
+    deleteat!(x, (n-g+1):n)
+    deleteat!(x, 1:g)
+    
+    return x
+end
+
+"""
+    winsor(x, p=0.2)
+
+Return a Winsorized version of vector `x`; i.e. replace the proportion
+`p` of the lowest elements of `x` with the next-lowest, and replace the
+proportion `p` of the highest elements with the previous-highest. This
+function is used to compute the Winsorized mean of `x`. It is also used
+by `trimvar` to compute the variance of the trimmed mean of `x`.
+
+# Example
+```julia
+julia> winsor([1,2,3,4,5], 0.2)
+ 2
+ 2
+ 3
+ 4
+ 4
```

Comment:
This is a pretty minor point, but it'd be nice to exactly emulate the output of `winsor` at the REPL. In particular, this is missing the `Array{Whatever,1}` stuff that gets printed before the list of elements.